src/zh/boylove/src/eu/kanade/tachiyomi/extension/zh/boylove/BoyLove.kt

```diff
@@ -37,18 +38,21 @@ class BoyLove : HttpSource(), ConfigurableSource {
 
     private val json: Json by injectLazy()
 
-    override val baseUrl = run {
-        val preferences: SharedPreferences =
+    override val baseUrl by lazy {
+        val preferences =
             Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
 
         val mirrors = MIRRORS
         val index = preferences.getString(MIRROR_PREF, "0")!!.toInt().coerceIn(0, mirrors.size - 1)
         "https://" + mirrors[index]
     }
 
-    override val client = network.cloudflareClient.newBuilder()
-        .rateLimit(2)
-        .build()
+    override val client by lazy {
```

Comment:
Refrain from abusing lazy when initializing is cheap, since lazy will introduce thread-safe overhead.